### PR TITLE
Fix caching issue

### PR DESCRIPTION
**Pull request by WenzDaniel**
Fixes issues of plugin caching. Before we only loaded from cache if the target is already cached. If a target is not cached we are initializing all plugins again. This can lead to very long plugin registration times in fuse.

Additionally, I cleaned up the function a bit by mostly rearranging stuff to make this PR work.

### CODE REVIEW

**Comment by WenzDaniel** [comment]
Maybe some additional explanation: 
* I moved the code from the local get_plugin function into the new `__get_plugin` method.
* The dictionary which contains the list of plugins returned by _get_plugins is now created more explicitly than the old non_local_plugins dict.
* For cleanliness I moved the lineage definition into its own function.

**Review by dachengx — CHANGES_REQUESTED** [review]
Thanks @WenzDaniel . The cleaning looks cool. I am still testing the PR to see any potential lineage change, but the comment is below.

**Comment by WenzDaniel** [comment]
Somehow I was just unable to find raw recrods aqmon and aqmon nv. When using this branch... need to check why first

**Comment by HenningSE** [comment]
Thanks for the work Daniel! I'm wondering if we can still increase the speed somehow. When working with fuse and also registering cuts strax still needs about 2-10 minutes to figure out how to load the requested cut data.

**Comment by WenzDaniel** [comment]
Fixed lineage change of raw records with https://github.com/AxFoundation/strax/pull/775

File: strax/context.py [hunk 1]
```diff
@@ -730,98 +724,118 @@ def _get_plugins(
...
 
-        # Initialize plugins for the entire computation graph
-        # (most likely far further down than we need)
-        # to get lineages and dependency info.
-        def get_plugin(data_type):
-            nonlocal non_local_plugins
+        plugins = {}
+        targets = list(targets)
+        counter = 0
+        while targets and counter < 100:
```
[the remark this answers is not in the document]
Comment (WenzDaniel) [reply strax/context.py]: @dachengx I increased the safety counter to 10_000 and added a raise condition if not all targest are initialized.

**Comment by WenzDaniel** [comment]
I guess we can ignore the complex method issue of codefactor I am not going to change that.

**Review by dachengx — APPROVED** [review]
Thanks @WenzDaniel .

**Pull request merged**